### PR TITLE
fix: add return values to prevent default on handled keys in the keyDown handler within the tree item component

### PR DESCRIPTION
**Pull request by tech4GT**
Signed-off-by: tech4gt <varun.gupta1798@gmail.com>

# Description

## Motivation & context

#3953 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Review by scomea — APPROVED** [review]
We should be returning false whenever we use the event (arrow keys + enter) and false otherwise.  One way would be for each case except the default one to return false.

File: packages/web-components/fast-foundation/src/tree-item/tree-item.ts
Comment (radium-v) [inline packages/web-components/fast-foundation/src/tree-item/tree-item.ts:200]: `return false` prevents default, stops propagation, and stops any further function execution.

Since this is the end of the function anyway, it should either just `return` (void) or not have a final return statement.

Comment (tech4GT) [reply packages/web-components/fast-foundation/src/tree-item/tree-item.ts:200]: Ah I thought of that, but ended up adding the statement anyway. I'll remove it! Thanks! :D

Comment (EisenbergEffect) [reply packages/web-components/fast-foundation/src/tree-item/tree-item.ts]: Here's the piece of code you all want to look at: https://github.com/microsoft/fast/blob/master/packages/web-components/fast-element/src/templating/binding.ts#L368

Comment (radium-v) [reply packages/web-components/fast-foundation/src/tree-item/tree-item.ts]: Well this is embarrassing 😳 the `return false` behavior I described is only relevant in the context of jQuery's `.on()` events, and may have been a thing for really old browsers (like, pre-2009). Returning false in a current-day event handler does nothing special. Either way, removing it here resolves exactly this sort of ambiguity, which is always good!

**Review by radium-v — APPROVED** [review]

File: packages/web-components/fast-foundation/src/tree-item/tree-item.ts
Comment (awentzel) [inline packages/web-components/fast-foundation/src/tree-item/tree-item.ts:189]: If we remove prevent default, should we also remove the comment for it in the line above?

Comment (tech4GT) [reply packages/web-components/fast-foundation/src/tree-item/tree-item.ts:189]: I left it in so as to not remove the information that it in fact prevents default (or returns true now) to prevent the page from scrolling. Do you think probably modifying it might be a better option?

Comment (tech4GT) [reply packages/web-components/fast-foundation/src/tree-item/tree-item.ts]: Updated the comment :D

**Comment by scomea** [comment]
This needs "Run 'yarn change' to generate a change file" done before it can be merged.

**Comment by tech4GT** [comment]
> This needs "Run 'yarn change' to generate a change file" done before it can be merged.

@scomea Shall I run it or, would it be run on master and I should wait for that?

**Comment by scomea** [comment]
@tech4GT Yes, you should run it locally and update the branch, after that I'll get it in.  (I'll click GitHub's "Update Branch" buttons to get a branch mergable, but beyond that I leave it to the change authors)

**Comment by EisenbergEffect** [comment]
Ping @scomea as a reminder on final review and merge of this fix.

**Review by EisenbergEffect — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
@scomea @chrisdholt Looks like this is fully approved and has been sitting here waiting to be merged for a long while. Is this still applicable and can it be merged?

**Comment by scomea** [comment]
> @scomea @chrisdholt Looks like this is fully approved and has been sitting here waiting to be merged for a long while. Is this still applicable and can it be merged?

No longer applicable.  Tree item does not even handle key events now.  Can be closed IMO.

**Comment by chrisdholt** [comment]
> > @scomea @chrisdholt Looks like this is fully approved and has been sitting here waiting to be merged for a long while. Is this still applicable and can it be merged?
> 
> No longer applicable. Tree item does not even handle key events now. Can be closed IMO.

Yep, this was one reason this was sitting. We had a “breaking” change to move the events wholesale.

**Pull request closed without merging**
